JavaFXUserInterface/src/main/java/SystemInfoController.java

```diff
@@ -224,4 +266,61 @@ void deleteDialog(MouseEvent mouseEvent) {
             uiUtilities.changeScene(mouseEvent, "/Systems");
         }
     }
+
+    public void generateRawDataTable() {
+        TableView table = new TableView();
+
+        // When TableRow is clicked, send data to SystemInfo scene.
+        table.setRowFactory(tv -> {
+            TableRow<Asset> row = new TableRow<>();
+            row.setOnMouseClicked(event -> {
+                uiUtilities.changeScene(event, row, "/SystemInfo", row.getItem());
+            });
+            return row;
+        });
+
+        TableColumn cycle = new TableColumn(CYCLE_COL);
+
+        TableColumn op1 = new TableColumn(OP1_COL);
+        TableColumn op2 = new TableColumn(OP2_COL);
+        TableColumn op3 = new TableColumn(OP3_COL);
+
+        TableColumn s1 = new TableColumn(S1_COL);
+        TableColumn s2 = new TableColumn(S2_COL);
+        TableColumn s3 = new TableColumn(S3_COL);
+        TableColumn s4 = new TableColumn(S4_COL);
+        TableColumn s5 = new TableColumn(S5_COL);
+        TableColumn s6 = new TableColumn(S6_COL);
+        TableColumn s7 = new TableColumn(S7_COL);
+        TableColumn s8 = new TableColumn(S8_COL);
+        TableColumn s9 = new TableColumn(S9_COL);
+        TableColumn s10 = new TableColumn(S10_COL);
+        TableColumn s11 = new TableColumn(S11_COL);
+        TableColumn s12 = new TableColumn(S12_COL);
+        TableColumn s13 = new TableColumn(S13_COL);
+        TableColumn s14 = new TableColumn(S14_COL);
+        TableColumn s15 = new TableColumn(S15_COL);
+        TableColumn s16 = new TableColumn(S16_COL);
+        TableColumn s17 = new TableColumn(S17_COL);
+        TableColumn s18 = new TableColumn(S18_COL);
+        TableColumn s19 = new TableColumn(S19_COL);
+        TableColumn s20 = new TableColumn(S20_COL);
+        TableColumn s21 = new TableColumn(S21_COL);
+        TableColumn s22 = new TableColumn(S22_COL);
+        TableColumn s23 = new TableColumn(S23_COL);
+        TableColumn s24 = new TableColumn(S24_COL);
+        TableColumn s25 = new TableColumn(S25_COL);
+        TableColumn s26 = new TableColumn(S26_COL);
+
+
+        table.setId("RawDataTable");
+        table.getColumns().addAll(cycle, op1, op2, op3, s1,s2,s3,s4,s5,s6,s7,s8,s9,s10,
+                s11, s12, s13, s14, s15, s16, s17, s18, s19, s20, s21, s22, s23, s24, s25, s26);
```

Comment:
same as my other comment, this is not flexible enough, don't hard code the sensors directly into the code but dynamically generate them from the info stored in the db

Reply:
You need to loop over the sensor data, and create the table columns dynamically.